unit-tests/fortran_intrinsic_module_test.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+##############################################################################
+# (c) Crown copyright 2018 Met Office. All rights reserved.
+# The file LICENCE, distributed with this code, contains details of the terms
+# under which the code may be used.
+##############################################################################
+"""
+Test of the rule for missing exit labels.
+"""
+from typing import List
+
+import pytest  # type: ignore
+from _pytest.fixtures import FixtureRequest  # type: ignore
+
+from stylist.fortran import IntrinsicModule
+from stylist.source import FortranSource, SourceStringReader
+
+
+# ToDo: Obviously we shouldn't be importing "private" modules but until pytest
+#       sorts out its type hinting we are stuck with it.
+
+
+class TestIntrinsicModule(object):
+    """
+    Tests the checker of missing exit labels.
+    """
+
+    @pytest.fixture(scope='class',
+                    params=["iso_c_binding", "ISO_Fortran_env",
+                            "ieee_exceptions", "ieee_arithmetic",
+                            "ieee_features", "other_module"])
+    def module_name(self, request: FixtureRequest) -> str:
+        """
+        Parameter fixture giving module names
+        """
+        return request.param
+
+    def test_exit_labels(self,
+                         module_name: str) -> None:
+        """
+        Checks that the rule reports missing "implicit" labels correctly
+        """
+        template = '''
+program test
+contains
+    function function1()
+        use {module_name}
+    end function function1
+
+    function function2()
+        use, intrinsic :: {module_name}
+    end function function2
+end program test
+'''
```

Comment:
Not catching all possible cases, e.g. top level program unit usage, but some of these are covered in the system testing.

Reply:
I don't think these are super necessary for a unit test - we're not testing fparser's implementation of `find_all` here.

Reply:
No but we could throwing some additional testing at our own `find_all`. https://github.com/MetOffice/stylist/blob/f857c000424914ccf5659ebe5227fd34ff8954b1/source/stylist/source.py#L324